src/main/webapp/app/lti/lti.route.ts

```diff
@@ -27,7 +28,14 @@ export const ltiLaunchRoutes: Routes = [
         },
     },
     {
-        path: 'deep-linking/:courseId',
+        path: 'deep-linking',
+        component: LtiCoursesComponent,
+        data: {
+            pageTitle: 'artemisApp.lti13.selectCourse',
```

Comment:
I am not 100% sure if this resolves your problem, but you are missing authorities and the `canDeactivate` prop here. (Does the header work correctly for the `Lti13DeepLinkingComponent` specified below?)

You can probably copy some of the configuration of the Course Managament route:
https://github.com/ls1intum/Artemis/blob/3f0878371d18ccba1d991febe15254b142f280fc/src/main/webapp/app/course/manage/course-management.route.ts#L27-L35

Reply:
Thank you so much for your help! Adding `canActivate: [UserRouteAccessService]` fixed my issue.
Actually, the header is working for `Lti13DeepLinkingComponent`.